test/jdk/java/awt/print/PrinterJob/PrintLatinCJKTest.java

```
@@ -31,59 +31,61 @@
+            """
+            + TEXT + """
+
 
             To test 8022536, if a remote printer is the system default,
             it should show in the dialog as the selected printer.
             """;
 
-    public static void showFrame() throws InterruptedException, InvocationTargetException {
-        invokeAndWait( () -> {
-            frame = new JFrame("Test Frame");
-            JButton b = new JButton("Print");
-            b.addActionListener((ae) -> {
-                try {
-                    PrinterJob job = PrinterJob.getPrinterJob();
-                    job.setPrintable(testInstance);
-                    if (job.printDialog()) {
-                        job.print();
-                    }
-                } catch (PrinterException ex) {
-                    ex.printStackTrace();
+    private static JComponent createTestUI() {
+        JButton b = new JButton("Print");
+        b.addActionListener((ae) -> {
+            try {
+                PrinterJob job = PrinterJob.getPrinterJob();
+                job.setPrintable(new PrintLatinCJKTest());
+                if (job.printDialog()) {
+                    job.print();
                 }
-            });
-            frame.getContentPane().add(b, BorderLayout.SOUTH);
-            frame.pack();
-
-            // add the test frame to dispose
-            PassFailJFrame.addTestWindow(frame);
-
-            // Arrange the test instruction frame and test frame side by side
-            PassFailJFrame.positionTestWindow(frame,
-                    PassFailJFrame.Position.HORIZONTAL);
-            frame.setVisible(true);
+            } catch (PrinterException ex) {
+                ex.printStackTrace();
+                String msg = "PrinterException: " + ex.getMessage();
+                JOptionPane.showMessageDialog(b, msg, "Error occurred",
+                                              JOptionPane.ERROR_MESSAGE);
+                PassFailJFrame.forceFail(msg);
+            }
```

> Displayed an error message to the tester if PrinterException is caught and automatically failed the test.

With the earlier test this failure was hidden until test timeouts and easy to miss, this code to handle PrinterException and forceFail the test immediately is a good addition.

---

This is why I insist on handling `PrinterException` or another possible exceptions in #17607, #17608, #17609 rather than swallowing them.